### PR TITLE
Add resizable column in table

### PR DESCRIPTION
**Pull request by frozenhelium**
## Changes

* Add option for column resize in table
* Remove first and last border in table

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] UI changes

### CODE REVIEW

**Comment by AdityaKhatri** [comment]
@frozenhelium ,
1. There should be a minimum width and whenever items overlap, overlapped content should not be shown
2. Make resize border larger

**Review by priyesh777 — APPROVED** [review]

**Comment by frozenhelium** [comment]
@AdityaKhatri 

1. Minimum width is possible through css `min-width` (Example in the story)
2. Resize border is actually larger (triggers on 4px) but it appears 1px visually. Should we make it visually large as well?

File: src/components/Table/useColumnResize.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            let newColumn = { ...column };
+
+            newColumn = {
+                ...column,
+                columnWidth: width ?? column.columnWidth,
+            };
+
+            return newColumn;
+        }, [])
+    ), [columns, columnWidths]);
```
Comment (tnagorra) [inline src/components/Table/useColumnResize.ts]: I think we shouldn't do this.

File: src/components/Table/index.tsx [hunk 2]
```diff
@@ -87,12 +93,32 @@ function Table<D, K extends string | number, C extends Column<D, K, any, any>>(
...
+                [name]: width,
+            }));
+        }
+    }, []);
+
+    const width = React.useMemo(() => (
+        sum(columns.map(
+            (column) => columnWidths[column.id] ?? column.columnWidth ?? DEFAULT_COLUMN_WIDTH,
+        ))
+    ), [columns, columnWidths]);
```
Comment (tnagorra) [inline src/components/Table/index.tsx]: How about we create a separate mapping with all of the defaults?
Currently there are ways the calculation of width may be inconsistent.

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Comment by tnagorra** [comment]
Let's also add default min-size on columns.
Let's also add background color so that the contents don't overlap with each other when column sizes are too small

**Pull request merged**